src/dotnet/APIView/APIView/Model/TokenTreeModel.cs

```diff
@@ -0,0 +1,182 @@
+            token.Kind = StructuredTokenKind.Content;
+            return token;
+        }
+
+        public static StructuredToken CreateSpaceToken()
+        {
+            var token = new StructuredToken();
+            token.Kind = StructuredTokenKind.NoneBreakingSpace;
+            return token;
+        }
+
+        public static StructuredToken CreateParameterSeparatorToken()
+        {
+            var token = new StructuredToken();
+            token.Kind = StructuredTokenKind.ParameterSeparator;
+            return token;
+        }
+
+        public static StructuredToken CreateTextToken(string value, string id = null)
+        {
+            var token = new StructuredToken(value);
+            if (!string.IsNullOrEmpty(id))
+            {
+                token.Id = id;
+            }
+            token.RenderClasses.Add("text");
+            return token;
+        }
+
+        public static StructuredToken CreateKeywordToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("keyword");
+            return token;
+        }
+
+        public static StructuredToken CreateKeywordToken(SyntaxKind syntaxKind)
+        {
+            return CreateKeywordToken(SyntaxFacts.GetText(syntaxKind));
+        }
+
+        public static StructuredToken CreateKeywordToken(Accessibility accessibility)
+        {
+            return CreateKeywordToken(SyntaxFacts.GetText(accessibility));
+        }
+
+        public static StructuredToken CreatePunctuationToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("punctuation");
+            return token;
+        }
+
+        public static StructuredToken CreatePunctuationToken(SyntaxKind syntaxKind)
+        {
+            return CreatePunctuationToken(SyntaxFacts.GetText(syntaxKind));
+        }
+
+        public static StructuredToken CreateTypeNameToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("type-name");
+            return token;
+        }
+
+        public static StructuredToken CreateMemberNameToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("member-name");
+            return token;
+        }
+
+        public static StructuredToken CreateLiteralToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("literal");
+            return token;
+        }
+
+        public static StructuredToken CreateStringLiteralToken(string value)
+        {
+            var token = new StructuredToken(value);
+            token.RenderClasses.Add("string-literal");
+            return token;
+        }
+    }
+
+    public class APITreeNode
+    {
+        public string Name { get; set; }
+        public string Id { get; set; }
+        public string Kind { get; set; }
```

Comment:
I see a null check for Kind below. Can this property be null anytime? Why don't we initialize a default value for these properties to reduces null check cost anytime we perform an operation on these tokens?

Reply:
Well it can be null, in which case I will like for the serializer to ignore the property. So its a bit of tradeoff

Reply:
We don't want to initial ta default. Because we want tit to be excluded form serilizatio if null.